ONSdigital/alpha-eq-survey-runner
# Digeq 55 setup interim save

**What**

This pull request creates the idea of a questionnaire session and the saving of interim data to a 
Cassandra cluster (enabled by https://github.com/ONSdigital/eq-terraform/pull/6) by creating a new 
REST service for auto save, implementing the client side js to enable this and then moving the deployment unit to docker container definitions.

In addition, this PR makes local development on eq-survey-runner easier by making use of docker-compose to create a full dev environment and enabling live reload from a local filesystem mounted inside the development container.

**How to test**
- **digeq-55**
  1. Create a environment based off of this branch, either using docker or the eq-terraform deployment.
  2. Create a new survey/ Questionnaire with questions.
  3. Start filling out a new questionnaire session.
  4. Stop halfway without submitting and reload that questionnaire session in another browser / incognito browser session.
  5. You should observe the data entered in the previous session.

***digeq-106**
@alexmorris to review :art: 
1. Checkout this branch.
2. Follow the instructions given in the README.md
3. Ensure that edits made to templates are reflected in the viewed output (live local webserver)

**Who can review**

For digeq-106 @alexmorris should review - for the other part anyone on the team but @dhilton
